### ISSUE TITLE
SMTP Account Setup Issue at 1&1 Webspace!

### ISSUE THREAD

**Issue by marius01** [owncloud/mail#673]
I have many IMAP Servers...STRATO...1und1 and Gmail :) No one works... not with autodiscover (without any extra data) and not with some configuration details like imap,smtp server (SSL)
It shows nothing if i dont type in any imap data... it just loads. if i config it with smtp and imap server it shows me the following: Cannot connect to SMTP. Its a Webspace (no Server), its an test application (no SSL), and we use an sqlite database.

**Comment by jancborchardt** [comment]
As said on IRC already we need more info from you to be able to properly debug this. Please ideally follow the [issue template](https://raw.githubusercontent.com/owncloud/core/master/issue_template.md)

At the very least we need some data from your owncloud.log and if the javascript error log of the browser has errors.

**Comment by TheChristian01** [comment]
### Steps to reproduce
1. Install Mail-App
2. Try to Setup for first time (with an Admin-Account)
3. If i use Auto-Discover (no additional IMAP Data and Ports) it loads for a long time and nothing happens. If i gave additional data, it tells me the following: "Cannot connect to SMTP-Server".

### Expected behaviour
I should configure the Mail-App, and start normally :) 

### Actual behaviour
If i try Autodiscover: It loads long time and nothing happens.
If i try Manual Configuration: It tells me, that the app cannot connect to SMTP. 

### Server configuration
Webspace on 1and1. Powerd by Linux.

**Web server:**
-

**Database**
SQLite

**PHP version:**
5.5

**ownCloud version:** (see ownCloud admin page)
8.0.3 (stable)

**Updated from an older ownCloud or fresh install:**
fresh!

**List of activated apps:**
Mail, Calendar, Contacts

**The content of config/config.php:**
nothing special!

```
Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)

```
run: sqlite3 data/owncloud.db
then execute:  select * from oc_appconfig where appid='user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**

**Operating system:**

### Logs
#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by jancborchardt** [comment]
Please add at least the owncloud.log and Browser log. :)

**Comment by TheChristian01** [comment]
wait some minutes :)

**Comment by TheChristian01** [comment]
I am sorry. I cannot upload them because there are too many private informations (e-Mail, Names etc.)

**Comment by enoch85** [comment]
@TheChristian01 Remove the private information in your post then. :)

**Comment by zinks-** [comment]
@TheChristian01 can you post the settings you use to successfully connect to your smtp from other mail clients ? (port, encryption settings, authentication type, etc).

**Comment by TheChristian01** [comment]
These are the following settings: 

imap.strato.de
SSL 
smtp.strato.de
SSL

E-Mail as Username 
Password

I try to setup Mail on a 1und1-Webspace. The Following Error is shown in the Log:

fsockopen(): unable to connect to smtpin.rzone.de:993 (Connection refused) at /homepages/3/d575768521/htdocs/apps/mail/lib/service/autoconfig.php#399

**Comment by irgendwie** [comment]
As I'm using a Strato Account for testing - I can confirm, that Strato Accounts **should work**.
@TheChristian01 your settings look correct for Strato, maybe it's just a certificate issue (no trust in CAs - often caused by missing "ca-certificates" package for debian / ubuntu based OS.)

**Comment by TheChristian01** [comment]
Hi. First. Strato is the E-Mail Postbox. 1und1 is the Webspace, where it is installed. Has anybody tested it with 1und1. Does not work with 1und1-Accounts and Gmail aswel.

Its a webspace, so im not able to change so many settings. Do you think i need an SSL Cert for my Domain?

**Comment by zinks-** [comment]
@TheChristian01 The error message comes from the autoconfig, and the autoconfig tries connecting to different ports to detect what works. It's therefore normal to have some logs that says some connection failed. If we do things the right way, the right combination *should* be found out at some point.

What happens if you enter the settings you pasted into the "manual configuration"?

@irgendwie Does your strato account work with autoconfig ?

**Comment by irgendwie** [comment]
@zinks- **no**

**Log:** ('[%DOMAIN%]' is my domain)
`{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"mail","message":"setting up auto detected account","level":1,"time":"2015-05-14T15:44:44+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}
{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"PHP","message":"fsockopen(): unable to connect to smtp.rzone.de:143 (Network is unreachable) at \/opt\/lampp\/htdocs\/oc\/apps\/mail\/lib\/service\/autoconfig.php#399","level":3,"time":"2015-05-14T15:44:45+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}
{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"PHP","message":"fsockopen(): unable to connect to smtp.rzone.de:585 (Network is unreachable) at \/opt\/lampp\/htdocs\/oc\/apps\/mail\/lib\/service\/autoconfig.php#399","level":3,"time":"2015-05-14T15:44:45+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}
{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"PHP","message":"fsockopen(): unable to connect to smtp.rzone.de:993 (Network is unreachable) at \/opt\/lampp\/htdocs\/oc\/apps\/mail\/lib\/service\/autoconfig.php#399","level":3,"time":"2015-05-14T15:44:45+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}
{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"PHP","message":"fsockopen(): unable to connect to [%DOMAIN%]:143 (Network is unreachable) at \/opt\/lampp\/htdocs\/oc\/apps\/mail\/lib\/service\/autoconfig.php#399","level":3,"time":"2015-05-14T15:44:45+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}
{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"PHP","message":"fsockopen(): unable to connect to [%DOMAIN%]:585 (Network is unreachable) at \/opt\/lampp\/htdocs\/oc\/apps\/mail\/lib\/service\/autoconfig.php#399","level":3,"time":"2015-05-14T15:44:45+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}
{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"PHP","message":"fsockopen(): unable to connect to [%DOMAIN%]:993 (Network is unreachable) at \/opt\/lampp\/htdocs\/oc\/apps\/mail\/lib\/service\/autoconfig.php#399","level":3,"time":"2015-05-14T15:44:45+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}
{"reqId":"VVTC7MCoGEcAABjVatgAAAAH","remoteAddr":"::1","app":"mail","message":"Auto detect failed","level":1,"time":"2015-05-14T15:44:45+00:00","method":"POST","url":"\/oc\/index.php\/apps\/mail\/accounts"}`

**Comment by TheChristian01** [comment]
It does not work with manual settings...I have tried it before :)

**Comment by zinks-** [comment]
OK, so autoconfig can't work with strato emails (at least those with custom domains not in ISPDB), because the MX record returns 'smtp.rzone.de', which is not the same server that users should use ('smtp.strato.de') to send mail.
The best advice to give strato users is to use manual config.

@TheChristian01 According to strato's documentation, STARTTLS on port 587 of smtp.strato.de should work as well. Can you try that in the manual config ?
The error message you get with manual config is the one we send when one of the parameters is wrong, so maybe using theses new ones will help.

**Comment by TheChristian01** [comment]
It does not work with Gmail or Outlook.com aswel. So i think it is an bug of owncloud mail.

**Comment by zinks-** [comment]
@TheChristian01 What kind of server have you installed owncloud on ? Have you been able to send any mail from that server using another program (alpine is easy to configure) ?
If you give me the name of your hosting provider I can maybe dig up some more info

**Comment by zinks-** [comment]
> So i think it is an bug of owncloud mail.
If we want to fix it, we must be able to reproduce it ourselves. That's why we're discussing here!

**Comment by TheChristian01** [comment]
I want to use it on my 1und1 Unlimited Plus Package. You can test it for 1 month, and can try it. 
It is based on Linux, but im not a root (its a webspace, no server). So i am not able to change many things. 
I am able to send emails with other apps (outlook, rainloop for owncloud...) but not with yours. (And i want to use yours because i think its great!!)

I have tested it on my STRATO Webspace. It works on this one! Your Mail app is great :)

I hope you can help me, to setup it on my 1und1 Pack :)

**Comment by zinks-** [comment]
> I am able to send emails with other apps (outlook, rainloop for owncloud...)
If you install rainloop on the same web space you installed our mail app (1&1 Unlimited +), you can send email? I'm asking confirmation because you threw outlook in the same sentence, and I don't think Microsoft has made a self-hosted version of their webmail client...

**Comment by TheChristian01** [comment]
1) Yes. Rainloop was installed on the same space. (as an app in owncloud)
2) I have tested it with the Windows Version of Outlook (of corse.)

But (Off-Topic): OWA (Outlook Web Access) is also an self-hosted Outlook client (powerd by ms exchange)

**Comment by zinks-** [comment]
@TheChristian01 I stand corrected on the outlook thing
OK so if Rainloop works in the same condition, the bug is confirmed (yay)

**Comment by TheChristian01** [comment]
So...What does that mean :)

**Comment by TheChristian01** [comment]
Can anybody test this at his own 1and1 Server? I hope you can fix this bug verry soon :)

**Comment by zinks-** [comment]
@TheChristian01 That means we'll work on it! I'll set up a 1&1 account and try to reproduce the bug, and if I can well, me or someone will try to fix it :)

**Comment by TheChristian01** [comment]
I hope anybody can :)

**Comment by TheChristian01** [comment]
I will try it now with an manual installation of owncloud. not websetup. Hope i can fix this :)

**Comment by zinks-** [comment]
> I will try it now with an manual installation of owncloud. not websetup. Hope i can fix this :)

Please tell us how this goes.

**Comment by TheChristian01** [comment]
It does not work :) Same problems :)

**Comment by Tribalize** [comment]
I can confirm the same errors. I use a 1and1 server. Can not connect with any email address. was using rainloop and it worked fine. I just tried to switch since the release of this app out of production. 

Errors i am seeing after trying to add accounts. Auto config times out when trying to conect to gmail
fsockopen(): unable to connect to alt1.gmail-smtp-in.l.google.com:993 (Connection timed out) at apps/mail/lib/service/autoconfig.php#399

Error after trying manual connect with 1and1 email.
Exception: {"Message":"Unable to generate a URL for the named route \"mail.page.index\" as such route does not exist.","Code":0,"Trace":"#0 \/homepages\/35\/d255602532\/htdocs\/public_html\/trib\/files\/lib\/private\/route\/router.php(282): Symfony\\Component\\Routing\\Generator\\UrlGenerator->generate('mail.page.index', Array, false)\n#1 
\/lib\/private\/urlgenerator.php(42): OC\\Route\\Router->generate('mail.page.index', Array)\n#2 \/apps\/mail\/appinfo\/app.php(21): OC\\URLGenerator->linkToRoute('mail.page.index')\n#3 \/lib\/private\/app.php(386): require('\/homepages\/35\/d...')\n#4 \/settings\/ajax\/navigationdetect.php(9): OC_App::getAppNavigationEntries('mail')\n#5 \/lib\/private\/route\/route.php(135) : runtime-created function(1): require_once('\/homepages\/35\/d...')\n#6 [internal function]: __lambda_func(Array)\n#7 \/lib\/private\/route\/router.php(250): call_user_func('\\x00lambda_118', Array)\n#8 \/lib\/base.php(782): OC\\Route\\Router->match('\/settings\/ajax\/...')\n#9 \/index.php(36): OC::handleRequest()\n#10 {main}","File":"\/3rdparty\/symfony\/routing\/Symfony\/Component\/Routing\/Generator\/UrlGenerator.php","Line":134}

**Comment by zinks-** [comment]
Just got confirmation from 1&1 support that port 465 and 587 are not blocked on their hosting.

**Comment by TheChristian01** [comment]
Can you fix this?

**Comment by TheChristian01** [comment]
Hi!
Im not longer interessted in fixing this bug. We changed to STRATO today. Everything works there :)
But I hope you can fix this sometime, to help all the 1and1-Costumers!

**Comment by Tribalize** [comment]
after uninstalling and reinstalling both this mail app and rainloop. this is what i have found.
this mail app and rainloop both can connect to servers with imap settings. 
Stmp is where the problem seems to exist. rainloop stmp only works by using php mail() function
This mail app will connect with imap but fails with stmp.

log below of gmail account and 1and1 email account

Error	mail	Creating account failed: Error connecting to SMTP server.	2015-05-19T18:06:59+00:00
Debug	mail	Connecting to smtp	2015-05-19T18:06:57+00:00
Debug	mail	Connecting to imap	2015-05-19T18:06:57+00:00
Info	mail	Setting up manually configured account	2015-05-19T18:06:57+00:00
Error	mail	Creating account failed: Error connecting to SMTP server.	2015-05-19T18:04:58+00:00
Debug	mail	Connecting to smtp	2015-05-19T18:04:56+00:00
Debug	mail	Connecting to imap	2015-05-19T18:04:56+00:00
Info	mail	Setting up manually configured account	2015-05-19T18:04:56+00:00
Error	mail	Creating account failed: Error connecting to SMTP server.	2015-05-19T18:04:52+00:00
Debug	mail	Connecting to smtp	2015-05-19T18:04:50+00:00
Debug	mail	Connecting to imap	2015-05-19T18:04:49+00:00
Info	mail	Setting up manually configured account

**Comment by zinks-** [comment]
@Tribalize thx for the followup! Can you find out what actual smtp server the mail() function uses?
I'm thinking it probably connects to 1&1's own smtp.
Maybe you can send it an email from rainloop  so that I can take a look at the headers (my address in on my profile page)?

Maybe the ports *are* blocked and either the customer support tried to lure a new customer (me) in, or they didn't understand the question (they were commercial support, not technical)

**Comment by marius01** [comment]
Hey. I will ask 1und1 Support about the Ports :)

**Comment by marius01** [comment]
I've just spoken with 1und1. These functions via the ports are actually for security reasons only possible with PHP and Sendmail.

**Comment by zinks-** [comment]
@marius01 ah thnaks for the info! I guess this might have just turned into a feature request then!
@jancborchardt @DeepDiver1975 Do we want to support setups where the only way of sending mail is through the mail() php function ?

**Comment by marius01** [comment]
Can anybody tell me the diffrence between SMTP and mail php? (disadvantages) ...

**Comment by zinks-** [comment]
@marius01 When you use mail(), the smtp server you send your message through is one decided by the hosting provider, not the one associated with your mail account.
this causes concerns for privacy, and  if you use [SPF](https://en.wikipedia.org/wiki/Sender_Policy_Framework) will cause problems (or will defeat its purpose if you have to add this public smtp server to the allowed senders)

**Comment by marius01** [comment]
That sounds bad for privacy :) Is there any other options?

**Comment by Tribalize** [comment]
After much research and testing including using http://mxtoolbox.com/ and telnet
i was able to connect 3 1and1 email accounts using stmp.1and1.com, stmp security none, port 25
But unable to add a gmail account. still getting stmp connect error there.

**Comment by jancborchardt** [comment]
@DeepDiver1975 @LukasReschke your comments on the above please ^

**Comment by clementhk** [comment]
mail() is native C (and through MTA), so it's a lot more reliable and fast. 

Clement

**Comment by marius01** [comment]
So...Is anybody able to fix this or make an workarround?

**Comment by marius01** [comment]
If i use php mail (): how can i config SPF-Records, to be proof, that nobody sends E-Mails from my Domain (only me and my owncloud users).

Can you integrate php mail() into 0.2 (when will this release come?)

Are there any other security problems with php mail (). I dont want to loose privacy and security :)

I hope anybody can help!

**Comment by zinks-** [comment]
@marius01

1. The 1&1 smtp is shared, so setting up SPF would "validate" any other mail sent by 1&1 client.
2. If you look closely, you will see that this issue has been tagged "0.2", so right now yes it is planned (that can change though)

Please try to refrain using these bug reports for questions about SPF, mail(), and anything not directly related to an issue in the mail app. You will find a lot of resources on the internet about those!

**Comment by teytey** [comment]
Same problem here "Erstellung des Kontos ist fehlgeschlagen:Error connecting to SMTP server."

Error	mail	Creating account failed: Error connecting to SMTP server.	2015-05-24T21:23:38+00:00
Error	mail	Creating account failed: Error connecting to SMTP server.	2015-05-24T21:23:14+00:00
Error	mail	Creating account failed: Error connecting to mail server.	2015-05-24T21:22:58+00:00

and 

Error	PHP	fsockopen(): unable to connect to mailgate.cs.fh-nuernberg.de:143 (Connection refused) at /homepages/29/d577489358/htdocs/owncloud/apps/mail/lib/service/autoconfig.php#399

**Comment by jancborchardt** [comment]
Closing in favor of the shorter issue https://github.com/owncloud/mail/issues/729 which contains just the solution to this issue. :) Thanl you @zinks-!
